File: go/config/Stk.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Stk 对应 config/Stk.py。
type Stk struct {
	*session.ApiGroup
}

// NewStk 创建 Stk API 分组。
func NewStk(s *session.Session) *Stk {
	return &Stk{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 stk/config.xml。
func (s *Stk) Config() (map[string]interface{}, error) {
	return s.S.Get("stk/config.xml", nil, "config")
}

File: src/config/stk.rs
//! Stk config group (`config/Stk.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Stk config group.
pub struct StkConfig<'a> {
    conn: &'a Connection,
}

impl<'a> StkConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        StkConfig { conn }
    }

    /// `stk/config.xml`. STK configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/stk/config.xml")
    }
}
